crates/biome_js_analyze/src/utils/restricted_glob.rs

```diff
@@ -98,12 +125,40 @@ impl schemars::JsonSchema for RestrictedGlob {
 /// Constructing candidates has a very small cost associated with it, so
 /// callers may find it beneficial to amortize that cost when matching a single
 /// path against multiple globs or sets of globs.
+#[derive(Debug, Clone)]
 pub struct CandidatePath<'a>(globset::Candidate<'a>);
 impl<'a> CandidatePath<'a> {
     /// Create a new candidate for matching from the given path.
     pub fn new(path: &'a impl AsRef<std::path::Path>) -> Self {
         Self(globset::Candidate::new(path))
     }
+
+    /// Tests whether the current path matches `glob` or not.
+    pub fn matches(&self, glob: &RestrictedGlob) -> bool {
+        glob.is_match_candidate(self)
+    }
+
+    /// Match against a list of globs where negated globs are handled as exceptions.
+    ///
+    /// This mimics the behavior of `.gitignore`.
+    /// For example, given the list `["**/*", "!a*", "aa", "!a"]`, we have:
+    /// - The file `a` doesn't match
+    /// - The file `aa` matches
+    /// - The file `aaa` doesn't match
+    /// - The file `b` matches
```

Comment:
This is **hard** to understand, really.

Here's a few pointers to make it simpler:
- don't mention gitignore, because some people might not know the feature you mentioned 
- the examples should explain **why they match/not match**. For example, I don't... And still can't 
- add doc tests, if you really want to have examples 
- you deferred the "exception" concept to gitignore, but didn't provide any link to it 

I suggest you explain the feature yourself, and avoid deleting to other resources